### ISSUE TITLE
Video length an hour too long

### ISSUE THREAD

**Issue by piiskop** [zdila/gpx-animator#346]
The rendered video has a length of an hour more than necessary. The final hour just shows the last frame.

**Comment by McPringle** [comment]
I have never seen this behavior. Can you please provide the saved configuration (`File > Save as`) together with the GPX file? You can upload it here to this issue or, if you prefer to keep it private, via email to support@gpx-animator.app (please add "Issue #346" to the subject).

With your files, I might be able to reproduce the problem and track down the error.

**Comment by piiskop** [comment]
Github doesn't support the [configuration file](https://nextcloud.tennis24.ee/index.php/s/JM7j3z7aNcX67ct) type.

**Comment by piiskop** [comment]
Github doesn't support [GPX-file](https://nextcloud.tennis24.ee/index.php/s/9g9TEPwn2gj3prQ) type.

**Comment by McPringle** [comment]
Create a ZIP file... ;-)

**Comment by piiskop** [comment]
I don't see why ZIP's are allowed and GPX's aren't. One has to zip and the other one has to unzip. I hate zipping-unzipping. Instead, I uploaded the file to my file sharing system for reusing and you can just click the link in my previous post and download it from there.

**Comment by McPringle** [comment]
Sorry for the long delay. I had some problems with my PC at home and needed a new power supply. Let's check your project:

![image](https://user-images.githubusercontent.com/1254039/116203544-18bf5200-a73c-11eb-85e6-e15a66d2e99c.png)

GPX Animator is doing exactly what you told him. My system is running in German, so GPX Animator is running in German, too. But with the red markings, it should be okay to follow my explanations to understand why you got what you got.

The first red marking shows you set the speedup to 1.0, which is realtime. A one hour recording will result in a one hour video. Do you really want that? It is very unusual, but will work fine.

The second red marking shows the default setting of a one hour tail duration. The tail duration is a separate color for the tail to show the movement. Together with your other setting of a realtime video this will result in an additional hour after finishing the recording, where you can see the tail color (black) animating to the track color (dark pink). This animation is hard to see because it is going so slowly (one hour).

Do you really want a 3.5 hour long video of your 2.5 hour long walk?

**Comment by piiskop** [comment]
I track my orienteering runs. Afterwards, I put them to my video that I have recorded yousing my action camera. This is why I need the track video to have the realtime duration as my running video also has it.

Having the tail with another color to differentiate the track I ran half an hour ago from the track I'm running currently is a nice feature especially in the conditions where I run through the same place many times. I want my ran distance to be visible for the whole video. Without a different tail color, it would be very difficult to tell which route I took lately. However, I don't want the tail color to be rendered after the run has been finished. There could be a setting that the user can set for ending the video in realtime or after the tail has a totally different color.
